controller-client/src/main/java/org/jboss/as/controller/client/impl/RemotingModelControllerClient.java

```diff
@@ -104,6 +105,9 @@ public void close() throws IOException {
             } finally {
                 StreamUtils.safeClose(clientConfiguration);
             }
+            // Per WFCORE-1573 remoting endpoints should be closed asynchronously, however consumers of this client
+            // likely need to wait until the endpoints are fully shutdown.
+            if (endpoint != null) endpoint.awaitClosedUninterruptibly();
```

Comment:
Should this just be endpoint.awaitClosed() with a catch for InterruptedException and conversion to InterruptedIOException, a la EnpdointImpl.close()?

Not being able to interrupt concerns me.